components/nexus-repository/src/main/java/org/sonatype/nexus/repository/search/SearchServiceImpl.java

```diff
@@ -162,7 +163,7 @@ public void put(final Repository repository, final Component component) {
         assets = Lists.newArrayList(tx.browseAssets(component));
       }
       String json = JsonUtils.merge(componentMetadata(component, assets), JsonUtils.from(additional));
-      client.get().prepareIndex(repository.getName(), TYPE, component.getEntityMetadata().getId().toString())
+      client.get().prepareIndex(repository.getName(), TYPE, identifier)
```

Comment:
minor, but there will be inevitable conflicts with https://github.com/sonatype/nexus-oss/pull/1211/files#diff-fce50f73841ede70a43ed2519c3addbe